src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -17,174 +17,744 @@
         {
-            JLOG (j_.error()) <<
-                "Malformed entry: '" << n << "'";
+            JLOG (j_.warn()) <<
+                "Invalid version from " << sites_[siteIdx].uri;
             return false;
         }
 
-        auto const id = parseBase58<PublicKey>(
-            TokenType::TOKEN_NODE_PUBLIC, match[1]);
+        auto manifest = Manifest::make_Manifest (beast::detail::base64_decode(
+            body["manifest"].asString ()));
 
-        if (!id)
+        ManifestDisposition result;
+        if (manifest)
+            result = manifests_.applyManifest (
+                Manifest (
+                    manifest->serialized, manifest->masterKey,
+                    manifest->signingKey, manifest->sequence),
+                *this);
+
+        if (! manifest || result == ManifestDisposition::invalid)
         {
-            JLOG (j_.error()) <<
-                "Invalid node identity: " << match[1];
+            JLOG (j_.warn()) <<
+                "Invalid manifest from " << sites_[siteIdx].uri;
+            return false;
+        }
+        else if (result == ManifestDisposition::untrusted)
+        {
+            JLOG (j_.warn()) <<
+                "Untrusted validator list signing public key (" <<
+                toBase58(TokenType::TOKEN_NODE_PUBLIC, manifest->masterKey) <<
+                ") from " << sites_[siteIdx].uri;
+            return false;
+        }
+        else if (manifest->revoked())
+        {
+            if (result == ManifestDisposition::accepted)
+            {
+                JLOG (j_.debug()) <<
+                    "Revoked validator list signing public key (" <<
+                    toBase58(
+                        TokenType::TOKEN_NODE_PUBLIC, manifest->masterKey) <<
+                    ") from " << sites_[siteIdx].uri;
+            }
             return false;
         }
 
-        if (trusted (*id))
+        pubKey = manifest->masterKey;
+        auto iter = publisherLists_.find (pubKey);
+        if (iter == publisherLists_.end ())
         {
             JLOG (j_.warn()) <<
-                "Duplicate node identity: " << match[1];
-            continue;
+                "Untrusted validator list signing public key (" <<
+                toBase58(TokenType::TOKEN_NODE_PUBLIC, pubKey) <<
+                ") from " << sites_[siteIdx].uri;
+            return false;
         }
 
-        if (insertPermanentKey(*id, trim_whitespace (match[2])))
-            ++count;
-    }
+        auto const blob =
+            beast::detail::base64_decode (body["blob"].asString());
+        r.parse (blob, list);
 
-    JLOG (j_.debug()) <<
-        "Loaded " << count << " entries";
+        if (list.isMember("sequence") && list["sequence"].isInt() &&
+            list.isMember("validators") && list["validators"].isArray())
+        {
+            auto const sequence = list["sequence"].asInt ();
+            if (sequence <= iter->second.sequence)
+            {
+                JLOG (j_.debug()) <<
```

Comment:
`sequence == iter->second.sequence` should be normal. There has been no change since the last time we checked. I could split this into a check for `==` and a check for `<` and log `debug` and `warn` respectively.
The bad signature message below this should definitely be changed to `warn`.